### PR TITLE
Fixed bug with inaccurate calc

### PR DESCRIPTION
**Pull request by ValentinaKozlova**
## Previous Behavior
inacurate-calc

## New Behavior
inacurate-calc-fix

- Fixes #
#28727

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                        |    Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                  Change |
| :----------------------------------------------------------------------------------------------------------------------- | --------------------------: | --------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>priority-overflow</samp> <br /> <abbr title='bundle-size/overflow-manager.fixture.js'>createOverflowManager</abbr> |  `4.379 kB`<br />`1.722 kB` |  `4.515 kB`<br />`1.746 kB` | `136 B` <br />`24 B`  |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                          | `12.636 kB`<br />`4.679 kB` | `12.772 kB`<br />`4.705 kB` | `136 B` <br />`26 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.544 kB`<br />`18.216 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `208.115 kB`<br />`57.974 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |   `36.409 kB`<br />`12.01 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |     `6.48 kB`<br />`2.203 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d445789f33a50255a6a51548d7282a482bc78a0b'>d445789f33a50255a6a51548d7282a482bc78a0b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">66</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">59</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/Avatar-mount_base.html">612</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/Avatar-mount.html">600</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/Button-mount_base.html">295</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/Button-mount.html">292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/Field-mount_base.html">1041</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/Field-mount.html">1075</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProvider-mount_base.html">636</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProvider-mount.html">651</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-mount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-mount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">66</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">59</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">60</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/MakeStyles-mount_base.html">838</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/MakeStyles-mount.html">845</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/Persona-mount_base.html">1632</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/Persona-mount.html">1601</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/SpinButton-mount_base.html">1354</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28728/perf-test-react-components/SpinButton-mount.html">1326</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/priority-overflow/src/overflowManager.ts [hunk 1]
```diff
@@ -148,6 +136,16 @@ export function createOverflowManager(): OverflowManager {
     return offsetSize;
   };
 
+  const getClientSize = (el: HTMLElement) => {
+    if (sizeCache.has(el)) {
```
Comment (ling1726) [inline packages/react-components/priority-overflow/src/overflowManager.ts:140]: is it necessary to cache this? I only see the `clientSize` used once per update to get the available size of the container

Comment (ValentinaKozlova) [reply packages/react-components/priority-overflow/src/overflowManager.ts:140]: oh, I need to refactor this function. I forgot about this

Comment (ValentinaKozlova) [reply packages/react-components/priority-overflow/src/overflowManager.ts]: I refactored it. But caching is there for consistency

**Review by ling1726 — APPROVED** [review]

**Pull request merged**